pandas/core/groupby/generic.py

```diff
@@ -1569,6 +1574,220 @@ def func(df):
+        if self.axis == 1:
+            raise NotImplementedError(
+                "DataFrameGroupBy.value_counts only handles axis=0"
+            )
+
+        with self._group_selection_context():
+            df = self.obj
+
+            # Check for index rather than column grouping
+            index_grouping = self.grouper.groupings[0].name is None
+
+            # Try to find column names
+            if index_grouping:
+                keys: Any = []
+                remaining_columns = self._selected_obj.columns
+            elif isinstance(self._selected_obj, Series):
+                keys = [grouping.name for grouping in self.grouper.groupings]
+                remaining_columns = [self._selected_obj.name]
+            else:
+                if isinstance(self.keys, ops.BaseGrouper):
+                    keys = [grouping.name for grouping in self.keys.groupings]
+                elif isinstance(self.keys, str):
+                    keys = [self.keys]
+                else:
+                    keys = self.keys
+
+                remaining_columns = [
+                    key for key in self._selected_obj.columns if key not in keys
+                ]
+
+            if subset is not None:
+                remaining_columns = [key for key in subset if key not in keys]
+
+            if dropna:
+                df = df.dropna(subset=remaining_columns, axis="index", how="any")
+
+            # Add the remaining_column keys to the main grouper
+            main_grouper, _, _ = get_grouper(
+                df,
+                key=self.keys,
+                axis=self.axis,
+                level=self.level,
+                sort=self.sort,
+                mutated=self.mutated,
+                dropna=dropna,
+            )
+            groupings = list(main_grouper.groupings)
+            for key in remaining_columns:
+                grouper, _, _ = get_grouper(
+                    df,
+                    key=key,
+                    axis=self.axis,
+                    level=self.level,
+                    sort=self.sort,
+                    mutated=self.mutated,
+                    dropna=dropna,
+                )
+                groupings += list(grouper.groupings)
+
+            # Take the size of the overall columns
+            result = df.groupby(
+                groupings,
+                as_index=self.as_index,
+                sort=self.sort,
+                observed=self.observed,
+                dropna=dropna,
+            ).size()
+
+            # Change the nameof the size result
+            if self.as_index:
+                result.name = RESULT_NAME
+            else:
+                result.rename({"size": RESULT_NAME}, axis=1, inplace=True)
+
+            if normalize:
+                # Normalize the results by dividing by the original group sizes
+                indexed_group_size = df.groupby(
+                    main_grouper, sort=self.sort, observed=self.observed, dropna=dropna
```

Comment:
self.dropna here as well.

Reply:
Done

Reply:
I fixed this and added a test. Which failed! It seems that groupby.size() with muliple keys and nulls can produce a strange multiindex which fails to align propely. The code is in generic.py line 1751 and I am not very happy about it! (I just do the division in numpy and remake the series...)